lib/remote_system.rb

```diff
@@ -175,4 +184,42 @@ def remove_file(file)
       "Could not remove file '#{file}' on host '#{host}'.\nError: #{e}"
     )
   end
+
+  def get_cmd_options(opts = {})
+    command_options = []
+
+    if @ssh_port
+      if opts["ssh"]
+        command_options.push("-p")
+      elsif opts["scp"]
+        command_options.push("-P")
+      else
+        raise Machinery::Errors::MachineryError.new("You must set either :ssh or :scp as flag " \
+          "in get_cmd_options!")
+      end
+
+      command_options.push(@ssh_port.to_s)
+    end
+
+    if @ssh_identity_file
+      command_options.push("-i")
+      command_options.push(@ssh_identity_file)
+    end
+
+    command_options
+  end
+
+  def build_basic_ssh_command
```

Comment:
these two commands are so much alike that we could handle the differences by passing `:ssh` or `:scp` as an argument